src/neo/Network/P2P/TaskManager.cs

```diff
@@ -243,8 +261,13 @@ private void OnTimer()
                     if (TimeProvider.Current.UtcNow - kvp.Value > TaskTimeout)
                     {
                         session.IndexTasks.Remove(kvp.Key);
-                        session.TimeoutTimes++;
-                        AssignSyncTask(kvp.Key, session);
+                        if (++session.TimeoutTimes > 3)
```

Comment:
We never reduce `TimeoutTimes` ? what's happend if at the begining we have two, in one day we have one, and in a week, we have another one, we terminate the session, and the node could be healthy, we should expire his bad rates.

Reply:
Yes, maybe we can reduce TimeoutTimes by one every day.